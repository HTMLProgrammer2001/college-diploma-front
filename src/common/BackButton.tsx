import React from 'react';
import {Button} from 'react-bootstrap';
import {useHistory} from 'react-router-dom';


const BackButton: React.FC<{}> = () => {
	const history = useHistory();

	return (
		<Button variant="light" onClick={() => history.goBack()}>Назад</Button>
	);
};

export default BackButton;
