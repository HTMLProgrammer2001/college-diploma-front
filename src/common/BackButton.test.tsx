import React from 'react';
import {shallow} from 'enzyme';

import BackButton from './BackButton';

//mocks
const mockGoBack = jest.fn();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useHistory: () => ({
		goBack: mockGoBack
	})
}));


describe('Test back button', () => {
	it('Test rendering', () => {
		const backButton = shallow(<BackButton/>);
		expect(backButton.html()).toMatchSnapshot();
	});

	it('Test back button push', () => {
		const backButton = shallow(<BackButton/>);

		//emit click button
		backButton.simulate('click');

		//tests
		expect(mockGoBack).toBeCalledTimes(1);
	});
});
