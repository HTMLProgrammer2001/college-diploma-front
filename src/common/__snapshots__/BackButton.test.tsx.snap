// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test back button Test rendering 1`] = `"<button type=\\"button\\" class=\\"btn btn-light\\">Назад</button>"`;
